File: src/main/java/features/console/ActionResolver.java
package features.console;

import java.util.function.Consumer;
import java.util.stream.Collectors;

public class ActionResolver {

    public static Consumer<ConsoleContext> getAction(CommandAction action) {
        switch (action) {
            case PrintHelp:
                return ActionResolver::printHelp;
            case AddBlockToPipeline:
                return ActionResolver::addBlock;
            default:
                throw new ConsoleException(String.format("Action: %s not supported", action));
        }
    }

    private static void addBlock(ConsoleContext consoleContext) {

    }

    public static void printHelp(ConsoleContext ctx)
    {
        var result = CommandParser.commands.stream().map(x -> x.name + " - " + x.description + System.lineSeparator()).collect(Collectors.joining());
        System.out.print(result);
    }
}

File: src/main/java/features/console/ActionResolver.kt
package features.console

import java.util.function.Consumer

object ActionResolver {

    fun getAction(action: CommandAction): Consumer<ConsoleContext> {
        when (action) {
            CommandAction.PrintHelp -> return Consumer { printHelp(it) }
            CommandAction.AddBlockToPipeline -> return Consumer { addBlock(it) }
            else -> throw ConsoleException(String.format("Action: %s not supported", action))
        }
    }

    private fun addBlock(consoleContext: ConsoleContext) {

    }

    fun printHelp(ctx: ConsoleContext) {
        val result = CommandParser.commands.map { x -> x.name + " - " + x.description + System.lineSeparator() }.joinToString { x -> x }
        print(result)
    }
}